### PR TITLE
Add function to tell whether a TypedArray underlying buffer is shared

### PR DESCRIPTION
**Pull request by ferjm**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/438)

### CODE REVIEW

File: src/typedarray.rs [hunk 1]
```diff
@@ -206,6 +207,14 @@ impl<T: TypedArrayElement, S: JSObjectStorage> TypedArray<T, S> {
         let (pointer, length) = self.data();
         slice::from_raw_parts_mut(pointer, length as usize)
     }
+
+    /// # Unsafety
+    ///
+    /// Return a boolean flag which denotes whether the underlying buffer
+    /// is a SharedArrayBuffer.
+    pub unsafe fn is_shared(&self) -> bool {
```
Comment (jdm) [inline src/typedarray.rs]: This does not need to be unsafe.

**Comment by jdm** [comment]
r=me with the unsafe removed.

**Comment by ferjm** [comment]
I knew I added it for a reason :)

```
❯ cargo build
warning: src/jsglue.cpp:586:5: warning: delete called on non-final 'JS::OwningCompileOptions' that has virtual functions but non-virtual destructor [-Wdelete-non-virtual-dtor]
warning:     delete static_cast<JS::OwningCompileOptions *>(aOpts);
warning:     ^
warning: 1 warning generated.
   Compiling mozjs v0.9.0 (file:///Users/ferjm/dev/mozilla/rust-mozjs)
error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
   --> src/typedarray.rs:214:9
    |
214 |         JS_GetTypedArraySharedness(self.object.as_raw())
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
    |
    = note: consult the function's documentation for information on how to avoid undefined behavior

error: aborting due to previous error

For more information about this error, try `rustc --explain E0133`.
The following warnings were emitted during compilation:

warning: src/jsglue.cpp:586:5: warning: delete called on non-final 'JS::OwningCompileOptions' that has virtual functions but non-virtual destructor [-Wdelete-non-virtual-dtor]
warning:     delete static_cast<JS::OwningCompileOptions *>(aOpts);
warning:     ^
warning: 1 warning generated.

error: Could not compile `mozjs`.

To learn more, run the command again with --verbose.
```

**Comment by ferjm** [comment]
@jdm could you publish a new release with this change, please?

**Comment by jdm** [comment]
Done.

**Comment by ferjm** [comment]
Thanks!

**Pull request merged**